### DISCUSSION TITLE
How to hide ID column by default and should not allow user to hide or unhide

### DISCUSSION THREAD

**Discussion by SRGajjala** [KevinVandy/material-react-table#32] [General]
ID column can be used to identify row with primary key and I can pass to server side to identify row to edit

**Comment by KevinVandy** [comment]
You probably just don't want an ID column if you are never going to show it. You can still include an id in your data though and use it. You'll be able to access it in `row.original.id`. Also you could set that id as the official row.id with the `getRowId` prop set to `() => row.original.id`

**Reply by KevinVandy** [reply]
the selection docs talk a little about the `getRowId` prop https://www.material-react-table.com/docs/guides/row-selection#useful-row-ids

**Reply by SRGajjala** [reply]
thanks for the help, but I have more important thing is below one

How to use custom column filter function with columnFilterModeOptions #31
